ark/type/methods/base.ts

```diff
@@ -79,36 +102,98 @@ interface Type<out t = unknown, $ = {}>
 
 	describe(description: string): this
 
+	/**
+	 * Change the behavior when unknown properties are encountered
+	 * - `ignore`: ignore unknown properties (default)
+	 * - 'reject': disallow objects with unknown properties
+	 * - 'delete': clone the object and keep only known properties
+	 */
 	onUndeclaredKey(behavior: UndeclaredKeyBehavior): this
 
+	/**
+	 * Change the behavior when unknown properties are encountered
+	 * Includes the whole object, not just the immediate properties
+	 * - `ignore`: ignore unknown properties (default)
+	 * - 'reject': disallow objects with unknown properties
+	 * - 'delete': clone the object and keep only known properties
+	 */
 	onDeepUndeclaredKey(behavior: UndeclaredKeyBehavior): this
 
+	/**
+	 * Validate the data, throwing `type.errors` instance on failure
+	 * Unlike `assert`, this method has a typed argument
+	 * @example const ConfigT = type({ foo: "string" }); export const config = ConfigT.from({ foo: "bar" })
+	 */
 	from(literal: this["inferIn"]): this["infer"]
 
+	/**
+	 * Cast this type to a different type.
+	 * const branded = type(/^a/).as<`a${string}`>() // Type<`a${string}`>
+	 */
 	as<t = unset>(...args: validateChainedAsArgs<t>): instantiateType<t, $>
 
 	// brand<const name extends string, r = applyBrand<t, Predicate<name>>>(
 	// 	name: name
 	// ): instantiateType<r, $>
 
+	/**
+	 * ArkType of the input shape.
+	 * Does not include any morphs, but does include narrows.
+	 * @example const inputT = T.in
+	 */
 	get in(): instantiateType<this["inferBrandableIn"], $>
+	/**
+	 * ArkType of the output shape.
+	 * Does not include any morphs, but does include narrows.
+	 * @example const outputT = T.out
+	 */
 	get out(): instantiateType<this["inferIntrospectableOut"], $>
 
 	// inferring r into an alias improves perf and avoids return type inference
 	// that can lead to incorrect results. See:
 	// https://discord.com/channels/957797212103016458/1285420361415917680/1285545752172429312
+	/**
+	 * Intersect this type with another type.
+	 * May result in `never` Disjoint ArkType.
+	 * @example const intersection = type({ foo: "number" }).intersect({ bar: "string" }) // Type<{ foo: number; bar: string }>
+	 * @example const intersection = type({ foo: "number" }).intersect({ foo: "string" }) // Disjoint
+	 */
 	intersect<const def, r = type.infer<def, $>>(
 		def: type.validate<def, $>
 	): instantiateType<inferIntersection<t, r>, $> | Disjoint
 
+	/**
+	 * Intersect this type with another type.
+	 * If the intersection results in Disjoint (Type<never>), immediately throws an error.
```

Comment:
Intersect another `Type` definition, throwing an error if the result is unsatisfiable.